LaTeX/LaTeX Log.sublime-syntax

```diff
@@ -5,45 +5,101 @@ scope: text.log.latex
 
 first_line_match: "This is (pdf|pdfe)?TeXk?, Version "
 
+################################################################################
+
 contexts:
   main:
-    - match: ".*Warning:"
-      scope: invalid.deprecated
-    - match: '[^:]*:\d*:.*'
-      scope: invalid.deprecated
-    - match: .*Error|^!.*
-      scope: invalid.illegal
+    - include: log-entries
+    - include: file-paths
+    - include: under-over-full
+    - include: strings
+
+  under-over-full:
+    - match: (?:Overfull|Underfull)
+      scope: keyword.control.hyphenation.latex
+      push:
+        - meta_scope: meta.log.latex.hyphenation
+
+        - match: \[\]\n
+          scope: keyword.control.hyphenation.latex
+          pop: 1
+
+        - match: '[0-9]+\-\-[0-9]+'
+          scope: variable.parameter.hyphenation.latex2
+
+###[ File paths ]###############################################################
+
+  file-paths:
+    - include: function-file-paths
+    - include: tag-file-paths
+    - include: type-file-paths
+
+  function-file-paths:
     - match: .*\.sty
       scope: entity.name.function
-    - match: .*\.cls
-      scope: entity.name.type.class
+
+  tag-file-paths:
     - match: .*\.cfg
       scope: entity.name.tag.configuration
     - match: .*\.def
       scope: entity.name.tag.definition
-    - match: .*Info.*
+
+  type-file-paths:
+    - match: .*\.cls
+      scope: entity.name.type.class
+
+###[ Log entries ]##############################################################
+
+  log-entries:
+    - include: error-log-entries
+    - include: warning-log-entries
+    - include: info-log-entries
+    - include: fixme-log-entries
+
+  error-log-entries:
+    - match: '.*Error|^!.*'
+      scope: invalid.illegal
+
+  warning-log-entries:
+    - match: '.*Warning:'
+      scope: invalid.deprecated
+    - match: '[^:]*:\d*:.*'
+      scope: invalid.deprecated
+
+  info-log-entries:
+    - match: '.*Info.*'
       scope: comment.block.documentation
+
+  fixme-log-entries:
     - match: ".*FiXme:"
```

Comment:
Let's add this single change for consistency with the other log patterns. Thre rest looks good to me.

```suggestion
    - match: '.*FiXme:'
```